src/Microsoft.AspNet.Mvc.ModelBinding/ValueProviders/FormCollectionValueProvider.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Mvc.ModelBinding.Internal;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding
+{
+    /// <summary>
+    /// Represents a value provider that contains the parsed form values.
+    /// </summary>
+    public class FormCollectionValueProvider :
+        MetadataAwareValueProvider<IFormDataValueProviderMetadata>,
+        IEnumerableValueProvider
+    {
+        private readonly CultureInfo _culture;
+        private PrefixContainer _prefixContainer;
+        private readonly Func<Task<IFormCollection>> _valuesFactory;
+        private IFormCollection _values;
+
+        /// <summary>
+        /// Creates a <see cref="FormCollectionValueProvider"/> wrapping
+        /// an existing set of <see cref="IFormCollection"/>.
+        /// </summary>
+        /// <param name="values">The key value pairs to wrap.</param>
+        /// <param name="culture">The culture to return with ValueProviderResult instances.</param>
+        public FormCollectionValueProvider([NotNull] IFormCollection values, CultureInfo culture)
+        {
+            _values = values;
+            _culture = culture;
+        }
+
+        /// <summary>
+        /// Creates a <see cref="FormCollectionValueProvider"/> wrapping
+        /// an existing set of <see cref="IFormCollection"/> provided by a delegate.
+        /// </summary>
+        /// <param name="valuesFactory">The delegate that provides the key value pairs to wrap.</param>
+        /// <param name="culture">The culture to return with ValueProviderResult instances.</param>
+        public FormCollectionValueProvider([NotNull] Func<Task<IFormCollection>> valuesFactory,
+                                                     CultureInfo culture)
+        {
+            _valuesFactory = valuesFactory;
+            _culture = culture;
+        }
+
+        public CultureInfo Culture
```

Comment:
`public CultureInfo Culture { get; }`. Assign this property in the ctor, we shouldn't need the backing readonly field.